go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -789,14 +790,9 @@ func (bnc *BaseNetworkController) isLocalZoneNode(node *kapi.Node) bool {
 	return util.GetNodeZone(node) == bnc.zone
 }
 
-// getActiveNetworkForNamespace returns the active network for the given namespace
-// and is a wrapper around util.GetActiveNetworkForNamespace
-func (cnci *CommonNetworkControllerInfo) getActiveNetworkForNamespace(namespace string) (util.NetInfo, error) {
-	var nadLister nadlister.NetworkAttachmentDefinitionLister
-	if util.IsNetworkSegmentationSupportEnabled() {
-		nadLister = cnci.watchFactory.NADInformer().Lister()
-	}
-	return util.GetActiveNetworkForNamespace(namespace, nadLister)
+// and is a wrapper around GetActiveNetworkForNamespace
+func (bnc *BaseNetworkController) getActiveNetworkForNamespace(namespace string) (util.NetInfo, error) {
+	return bnc.nadController.GetActiveNetworkForNamespace(namespace)
```

Comment:
I am going to flag some use cases where, if we had per-namespace NAD information in NetInfo, we probably wouldn't need access to the nad controller to query this information. This is one of them.

We would lose the protection of actually querying UDNs, true.

Just for awareness.

Reply:
I think this is a result of me changing this function to be a receiver for bnc rather than the common network controller (that doesn't have netInfo). However, it looks like this function is used to determine pod wiring for the default controller, so I think makes sense to have the UDN protection here.